discovery_engine_core/core/src/config.rs

```diff
@@ -193,9 +197,62 @@ impl Default for ExplorationConfig {
     }
 }
 
-/// Reads the DE configurations from json and sets defaults for missing fields (if possible).
+/// Configurations for the feed.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct FeedConfig {
+    /// The maximum number of documents per feed batch.
+    pub(crate) max_docs_per_batch: usize,
+}
+
+impl FeedConfig {
+    /// Merges existent values from the DE configuration into this configuration.
+    pub(crate) fn merge(&mut self, de_config: &Figment) {
+        if let Ok(max_docs_per_batch) = de_config.extract_inner("feed.max_docs_per_batch") {
+            self.max_docs_per_batch = max_docs_per_batch;
+        }
+    }
+}
+
+impl Default for FeedConfig {
+    fn default() -> Self {
+        Self {
+            max_docs_per_batch: 2,
+        }
+    }
+}
+
+/// Configurations for the search.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct SearchConfig {
+    /// The maximum number of documents per search batch.
+    pub(crate) max_docs_per_batch: usize,
+}
+
+impl SearchConfig {
+    /// Merges existent values from the DE configuration into this configuration
+    pub(crate) fn merge(&mut self, de_config: &Figment) {
+        if let Ok(max_docs_per_batch) = de_config.extract_inner("search.max_docs_per_batch") {
+            self.max_docs_per_batch = max_docs_per_batch;
+        }
+    }
+}
+
+impl Default for SearchConfig {
```

Comment:
:wrench:  (see above)

```suggestion
#[cfg(test)]
impl Default for SearchConfig {
```